frontend/src/components/studySamples/StudyStepSamplesTable.tsx

```diff
@@ -45,15 +46,44 @@ function StudyStepSamplesTable({ studyID, step, settings }: StudyStepSamplesTabl
 	const protocol: Protocol | undefined = protocolsByID[step.protocolID]
 	const stepDefinition = stepsByID[step.stepID]
 
+	const actionColumn = useMemo(() => ({
+		columnID: 'Action',
+		title: 'Action',
+		dataIndex: ['sample', 'id'],
+		render: (_: any, { sample }: SampleAndLibrary) => {
+			return <Popconfirm
+				title={`Are you sure you want to remove sample '${sample?.name ?? 'Loading...'}' from step '${step.stepName}'?`}
+				onConfirm={async () => {
+					if (!sample) return;
+					const REMOVE_NOTIFICATION_KEY = `StudyStepSamplesTable.remove-${studyID}-${step.stepID}-${sample.id}`
+					notification.info({
+						message: `Removing sample '${sample?.name}' from step '${step.stepName}'`,
+						key: REMOVE_NOTIFICATION_KEY
+					})
+					await dispatch(api.sampleNextStepByStudy.remove(step.sampleNextStepByStudyBySampleID[sample.id].id))
+					await dispatch(refreshStudySamples(studyID))
+					notification.close(REMOVE_NOTIFICATION_KEY)
+				}}
+				disabled={!sample}
+				placement={'topLeft'}
+			>
+				<Typography.Link underline type={'danger'} href={''}>Remove</Typography.Link>
+			</Popconfirm>
+		}
+	}), [dispatch, step.sampleNextStepByStudyBySampleID, step.stepID, step.stepName, studyID])
+
 	const columns: IdentifiedTableColumnType<SampleAndLibrary>[] = useMemo(() => {
 		if (protocol && stepDefinition) {
 			// Same columns as labwork, but we don't want the Project column, since the user
 			// is already in the project details page.
-			return getColumnsForStudySamplesStep(stepDefinition, protocol)
+			return [
+				...getColumnsForStudySamplesStep(stepDefinition, protocol),
+				actionColumn,
```

Comment:
Is it wise to bypass the function that return the columns ?

Reply:
@nafiz1001 Yes, this is fine. We should be able to add custom columns to our tables like this, beyond the predefined columns defined for specific data types.